### PR TITLE
AK+LibCore+NetworkServer+NetworkSettings: Port network-related things to new Strings

### PR DESCRIPTION
**Pull request by sppmacd**
- AK: Add JsonValue::to_string() convenience wrapper
- NetworkServer: Convert to new Strings
- LibCore: Add ArgsParser::add_positional_argument with Optional<String>
- NetworkSettings: Port to new Strings

### CODE REVIEW

File: AK/JsonValue.h [hunk 1]
```diff
@@ -90,6 +91,13 @@ class JsonValue {
             return as_string();
         return serialized<StringBuilder>();
     }
+
+    ErrorOr<String> to_string() const
+    {
+        if (is_string())
+            return String::from_deprecated_string(as_string());
+        return String::from_deprecated_string(serialized<StringBuilder>());
+    }
```
Comment (linusg) [inline AK/JsonValue.h:100]: No. New to_string functions should propagate errors all the way, this is no improvement over the existing `to_deprecated_string()`. Temporary conversions between the old and new types should happen at the call site.

**Review by linusg — CHANGES_REQUESTED** [review]

File: AK/JsonValue.h (hunk 1 above)
Comment (sppmacd) [reply AK/JsonValue.h:100]: > No. New to_string functions should propagate errors all the way, this is no improvement over the existing `to_deprecated_string()`. Temporary conversions between the old and new types should happen at the call site.

`String::from_deprecated_string` already handles errors, and getting rid of that would require actually porting all the JsonValue to String.

Comment (linusg) [reply AK/JsonValue.h:100]: > String::from_deprecated_string already handles errors

Not allocation errors though, which is ultimately the primary concern of the new string type. UTF-8 validation is a nice bonus, but secondary.

> and getting rid of that would require actually porting all of JsonValue to String.

Right, there's more work needed here than wrapping this in a `from_deprecated_string()` and calling it a day :)

Comment (sppmacd) [reply AK/JsonValue.h:100]: Ok, I will mark this as draft for now and port JsonValue to String separately (unless someone does this first)

**Comment by sppmacd** [comment]
this is not stale

**Pull request closed without merging**